Makefile

```diff
@@ -58,8 +58,15 @@ checksum:
 clean:
 	@rm -rf $(BINDIR) ./_dist
 
+GIT_BASEDIR    = $(shell git rev-parse --show-toplevel 2>/dev/null)
+ifneq ($(GIT_BASEDIR),)
+	LDFLAGS += -X github.com/Azure/acs-engine/pkg/test.JUnitOutDir=${GIT_BASEDIR}/test/junit
+endif
+
 test: test-style
-	go test -v $(GOFILES)
+	go get github.com/onsi/ginkgo/ginkgo
+	go get github.com/onsi/gomega
```

Comment:
Agreed and good points @jchauncey. At the moment we our own dev process doesn't have clear instructions  on developing with the Docker image, so de facto almost all development is happening outside of containers. In a subsequent issue/PR (not this one) let's flesh out how developing acs-engine in a container can be better.